src/Kunstmaan/MediaBundle/Helper/File/FileHandler.php

```diff
@@ -183,13 +200,28 @@ public function saveMedia(Media $media)
      */
     public function getOriginalFile(Media $media)
     {
-        $relativePath = sprintf(
-            '/%s.%s',
+        return $this->fileSystem->get($this->getFilePath($media), true);
+    }
+
+    /**
+     *
+     *
+     * @param Media $media
+     * @return string
+     */
+    private function getFilePath(Media $media)
+    {
+        $filename = $media->getOriginalFilename();
+
+        if (!empty($this->blacklistedExtensions)) {
+            $filename = preg_replace('/\.('.join('|', $this->blacklistedExtensions).')$/', '.txt', $filename);
+        }
+
+        return sprintf(
+            '/%s/%s',
```

Comment:
Yes, if you use fat32 then the limit of files in a folder is 65,534. This way you work arround it. If we would drop fat32 this was not needed :)